From: You-Sheng Yang <vicamo.yang@canonical.com>
Date: Fri, 11 Oct 2024 02:28:05 +0800
Subject: UBUNTU: SAUCE: backport: dma-mapping: don't return errors from
 dma_set_max_seg_size

Accommodate to v6.12-rc1 commit 334304ac2baca ("dma-mapping: don't
return errors from dma_set_max_seg_size").

Bug-Ubuntu: https://bugs.launchpad.net/bugs/2083997
Signed-off-by: You-Sheng Yang <vicamo.yang@canonical.com>
---
 drivers/media/pci/intel/ipu7/ipu7.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/media/pci/intel/ipu7/ipu7.c b/drivers/media/pci/intel/ipu7/ipu7.c
index 4beb2a1..fccefd3 100644
--- a/drivers/media/pci/intel/ipu7/ipu7.c
+++ b/drivers/media/pci/intel/ipu7/ipu7.c
@@ -23,6 +23,7 @@
 #include <linux/scatterlist.h>
 #include <linux/slab.h>
 #include <linux/types.h>
+#include <linux/version.h>
 #include <linux/vmalloc.h>
 #include <media/ipu-bridge.h>
 
@@ -1795,9 +1796,13 @@ static int ipu7_pci_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	if (ret)
 		return dev_err_probe(dev, ret, "Failed to set DMA mask\n");
 
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 12, 0)
 	ret = dma_set_max_seg_size(dev, UINT_MAX);
 	if (ret)
 		return dev_err_probe(dev, ret, "Failed to set max_seg_size\n");
+#else
+	dma_set_max_seg_size(dev, UINT_MAX);
+#endif
 
 	ret = ipu7_pci_config_setup(pdev);
 	if (ret)
